nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-site-to-site/src/main/java/org/apache/nifi/remote/StandardRemoteGroupPort.java

```diff
@@ -366,6 +371,17 @@ private int receiveFlowFiles(final Transaction transaction, final ProcessContext
         return flowFilesReceived.size();
     }
 
+    private String createTransitUri(final Transaction transaction, String sourceFlowFileIdentifier) {
+        // Use actual uri as transitUri if this transport protocol has meaningful uri.
+        switch (remoteGroup.getTransportProtocol()) {
```

Comment:
Hmm. Theoretically, I agree with that. However, there're couple of reasons I ended up writing code like that:

1. Existing Socket site-to-site transit URI has `sourceFlowFileIdentifier` in it, which can't be a part of the URI that the communicant knows. I think this can be removed since a provenance event has `sourceSystemFlowFileId`. If it's ok to change existing rule, then we can use communicant url as transit url here.

2. There're two URLs in this context, one is Communicant (Peer) url, another is CommunicationsSession's url. For RAW socket, both look the same as "nifi://{hostname}:{port}". For HTTP, Peer.url="http://{hostname}:{port}/nifi-api", CommunicationsSession.url="http://{hostname}:{port}/nifi-api/data-transfer/input-ports/{uuid}/transactions/{tx-id}/flow-files". I thought CommunicationsSession's url is more appropriate for transit url.

Having said that, I think those distinction shouldn't exist here in StandardRemoteGroupPort. I will move those to more protocol specific classes. Thanks @markap14 for your comment!